src/lib.rs

```diff
@@ -183,9 +186,24 @@ pub enum Target {
 /// [`Logger::new()`]: #method.new
 /// [`Builder`]: struct.Builder.html
 pub struct Logger {
+    writer: BufferWriter,
     filter: filter::Filter,
-    format: Box<Fn(&mut Write, &Record) -> io::Result<()> + Sync + Send>,
-    target: Target,
+    format: Box<Fn(&mut Formatter, &Record) -> io::Result<()> + Sync + Send>,
+}
+
+/// A formatter to write logs into.
+/// 
+/// `Formatter` implements the standard [`Write`] trait for writing log records.
+/// It also supports terminal colors, but this is currently private.
+/// 
+/// [`Write`]: https://doc.rust-lang.org/stable/std/io/trait.Write.html
+pub struct Formatter {
```

Comment:
I think I'll move this into a new `fmt` module, since when we do make colouring public there will probably be some more types to go along with it.

If you just create your closure as part of the `.format` call then you won't need to name the `Formatter` type directly.